### PR TITLE
Fix bug when selecting multiple values in AsyncAutocomplete

### PR DESCRIPTION
**Pull request by mattlong**
The `onChange` events emitted from `AsyncAutocomplete` were incorrect when multiple values are selected; only the most recently selected value wound up being included in the call to the `onChange` handler.

### CODE REVIEW

File: packages/mock/src/mocks/valueset.ts [hunk 1]
```diff
@@ -8,8 +8,18 @@ export const exampleValueSet: ValueSet = {
     contains: [
       {
         system: 'x',
-        code: 'test-code',
-        display: 'Test Display',
+        code: 'test-code-1',
```
Comment (codyebberson) [inline packages/mock/src/mocks/valueset.ts:11]: I *think* this is fine, but there's minor risk that a random customer test is relying on this `code` or `display`.  If we wanted to be really safe, we'd leave these as-is, and add the new codes.

I'm probably fine with how you've done it though.

**Review by codyebberson — APPROVED** [review]

File: packages/mock/src/mocks/valueset.ts (hunk 1 above)
Comment (mattlong) [reply packages/mock/src/mocks/valueset.ts:11]: Ah, good point. No reason for the churn. I'll switch this one back.

**Comment by dillonstreator** [comment]
Thoughts on including some of the additional fixes in this other PR? https://github.com/medplum/medplum/pull/4393

**Review by codyebberson — APPROVED** [review]

**Comment by codyebberson** [comment]
@mattlong - can you please take a look at @dillonstreator 's related PR? https://github.com/medplum/medplum/pull/4393

**Comment by mattlong** [comment]
@dillonstreator Thanks so much and apologies for missing your PR. I integrated most of the changes from your PR into this one.

**Pull request merged**